velox/exec/HashTable.h

```diff
@@ -751,7 +751,7 @@ class HashTable : public BaseHashTable {
   // or distinct mode VectorHashers in a group by hash table. 0 for
   // join build sides.
   int32_t reservePct() const {
-    return isJoinBuild_ ? 0 : 50;
+    return (isJoinBuild_ && rows_->nextOffset()) ? 0 : 50;
```

Comment:
Is this an unrelated fix? Maybe extract it into a separate PR and explain what it does.

Reply:
Currently, there is no variable in the HashTable that can represent whether duplicate data is allowed. However, rows_->nextOffset() can be used to indicate that duplicate data is not allowed. For example, in the HashTable of a left semi join, rows_->nextOffset() is 0. Introducing a new variable `allowDuplicates` in HashTable here may make it clearer.

I feel that if HashBuild builds a hash table while adding input rows, it should also reserve some values for new keys, just like agg. I'm not sure if my understanding of `reservePct` is correct, please correct me if I'm wrong.